netx.go

```diff
@@ -38,14 +86,26 @@ func DialUDP(network string, laddr, raddr *net.UDPAddr) (*net.UDPConn, error) {
 func DialTimeout(network string, addr string, timeout time.Duration) (net.Conn, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	conn, err := DialContext(ctx, network, addr)
+	if isNetworkUnreachable(err) {
+		addr = convertAddressDNS64(addr)
+		conn, err = DialContext(ctx, network, addr)
+	}
+
 	cancel()
 	return conn, err
 }
 
 // DialContext dials the given addr on the given net type using the configured
 // dial function, with the given context.
 func DialContext(ctx context.Context, network string, addr string) (net.Conn, error) {
-	return dial.Load().(func(context.Context, string, string) (net.Conn, error))(ctx, network, addr)
+	dialer := dial.Load().(func(context.Context, string, string) (net.Conn, error))
+
+	conn, err := dialer(ctx, network, addr)
+	if isNetworkUnreachable(err) {
+		addr = convertAddressDNS64(addr)
+		conn, err = dialer(ctx, network, addr)
```

Comment:
if this is called with network "tcp" or "udp" this seems fine.  If this was called explicitly with "tcp4" or "upd4" or "ip4" it's questionable that we should re-dial vs just return the error.  I think it may fail anyway since we don't redial with "tcp" or "tcp6" but may return a different error.